### ISSUE TITLE
In a hierarchical layout, how to constrain dimensions?

### ISSUE THREAD

**Issue by jylertones** [kieler/elkjs#292]
TLDR; for the lower two screenshots, how can I constrain the width of the parent node so that it doesn't stretch wider when the edges are pathed?

I'm trying to build a hierarchical layered model. For those interested, it portrays database schema information as it is replicated. As such, it has database tables and contain database columns. I want to show a database table and columns like this:

![image](https://github.com/user-attachments/assets/746cca7a-e0d3-4d7d-a367-5352f7e6c3b6)
Customer is the table, the others are columns inside of the table. For display, the table node is the hierarchical container for the columns, and they are the same width.

For some complex layouts, the edge bend points for the columns will go inside of the table node, thereby making the table wider than the columns and breaking the layout.

![image](https://github.com/user-attachments/assets/a157f0d2-1487-4f80-b96b-420656bc2396)

Or as seen in the Elk tool:
![image](https://github.com/user-attachments/assets/335157a2-f02f-4aca-a692-4ffe2c330d15)

Elk Layout (sorry, you'll have to copy this into the tool as the link would not work): [layout.json](https://github.com/user-attachments/files/16255952/layout.json)

**ELK Version**
Tested using 0.9.2 and 0.9.3

**Things I have tried**
I've tried various settings, but I've been unable to get the bend points to happen outside of the node. I've been unable to limit the width of the graph. I've also tried using ports and merging lines, but then I feel like I lose helpful information on the graph.

**Comment by soerendomroes** [comment]
Hi, this is not directly possible because, we layout (most times) bottom up, meaning that the inner content is layouted first such that the bends are produced at the innermost node.

I suggest using explicit ports and not using `hierarchyHandling: INCLUDE_CHILDREN` as shown [here](https://rtsys.informatik.uni-kiel.de/elklive/elkgraph.html?compressedContent=IYGw5g9gTglgLgCwLYC4AEJgE8CmUcAmAUEQHYQE5qkCMZFVpATGgN5FqdoAO0cAwhFIBnOFGAxScYegBiASQAaAUQAiAfQDyAJVXLtaAPSG00NADMYAD0I8Iw+DCEWz+UGlAwwpJDikcuXig4HiYWdi5ItAA9ILgAOklKK3Q6KK5MACMcEDQAIm4wvIDOAF8Su2DQgGY2CsjYvkTSZPQABnqM4GzcgqZq4sjyyPJKajCSEYZx6smuQjAqQpYAWgA+caY5zgWl-rR1maJy3eoaA43meOWiU9oL8ev+oA). Additionally, you might want to set all spacings correct and maybe set `portConstraints: FIXED_POS` and set the desired positions of ports.

Configuring the layout to layout top-down instead of bottom up might also work. Maybe @Eddykasp can make a minimal example for this and explain how to apply the scaling values for this.

**Comment by soerendomroes** [comment]
Maybe [this](https://rtsys.informatik.uni-kiel.de/elklive/elkgraph.html?compressedContent=IYGw5g9gTglgLgCwLYC4AEJgE8CmUcAmAUEQHYQE5qkCMZFVpATGgN5Fqdo4gDWAdAAdgBAjFJh0AIgDacCIIC8ABn7KANCBwAzOCrXqARhDjyk+jbDAI9q5QF0pHLgGdhAY3Fh+h4C5wAaqAArjjoys6cgtBwAMIQpC5wUMDicC7oAGIAkgAaAKIAIgD6APIASoX55WgA9LVo0GjaMAAehGjRLvAwCc1N+KBooDBgpEg4pHCRnTGdTCzsXMtoAHrRUHD84pSt6HQrXJiGPGhSggtOywC+Mxtw8wDMbDPL6zHbpLvhr0fAJyAzhdHlcuLdluRKNQFiQIQxoY9YVxCGAqBcWABaAB80KYSM4KLRTGe2IRRFuhOoNDQpOYQhhlNoNJxdOBQA) also a better example that additionally sets spacing and padding.

**Comment by soerendomroes** [comment]
I have another model [here](https://rtsys.informatik.uni-kiel.de/elklive/elkgraph.html?compressedContent=IYGw5g9gTglgLgCwLYC4AEJgE8CmUcAmAUEQPSloDCCwAdmDmoowM4wBejEAZmrQIxMIaUCAgB3NCzhRgMMAjhpCDFgDoitCARwAFTAGMcSHLThrpsuDjBZ0AOQCiAFQDqAeQBKAaQD6AZQBJAFldABlHAA1NbT1DY1NzWhw4cWgAa38YJAAHEBwADzUtHQAxfIKYACMYEHg7NHt3ABFHAMCALUcYnX1gIxMzYpS0qEzsvMLi2PLC6tr6tR1uYABXEDgHFragrqIc6DgWf1WoKAhV2gIYenQAIgBtOAgcgF4AVjUABgAafO44B9vj8qhA4M8kEDfrAFIDPl8ALp3EglRgCNAAbyIAF8emiAEyYohoEkicDQeDIdD4AxwHL9dI3MDE0nAFg5HC0zzAOAwCDoL5qd5ochoACqLEYiBgLDQ0DQSGwVVYKRETGAUAYcFcMAIiCEaHpBnSTAQjBpcDoYHysogtCEOTlvBw-QQcuYUA0pPVmpSOr1CHQABYviySTgQOk1PSCNdbmhHs83oLfv9ASmQWCIa8MzDFDnvkiw3xYvZYgLiwcoHBKHbLHIzCx0KVApFHM1fLp3P4RRR5dwYAVCIaIGxeXa0Nx5fhQGT5LRBnBK4dDfjCVjvd7MFgLkoHgcx3zaAKfmh+PihQjewrgGAYAZi96AHpV8w3HQFdD8R+kl+HCy6jg6CuI4-jOD+JKYMqIAJjka7It6uLeq+q4AMxEpupLbruaD7qO8BHieaBBhe7xXqKip3g+mEkn+1ZqO+hQVjRaB0eYbA6MBoHgSxUERrB+KoQhpJIaSqJ8GuJDeuJtCCTiRAqGiggALQAHwSdGkmKXwKnqbJmmoUAA) that uses a few different configurations that might be interesting to try out.

**Comment by soerendomroes** [comment]
Or do you want something like [this](https://rtsys.informatik.uni-kiel.de/elklive/elkgraph.html?compressedContent=IYGw5g9gTglgLgCwLYC4AEJgE8CmUcAmAUEQHYQE5qkCMZFVpATGgN5Fqdo4gDWAdAAdgBAjFJh0AIgDacCIIC8ABn7KANCBwAzOCrXqARhDjyk+jbDAI9q5QF0pHLuUoBlGAC8c-AMYRSAGc4KGBxOED0ADkAeQARAFEAfQAZAEEAIQSUt2dOVxwU4EMeQKFMXxwkHFI4aQAJJIBhBKiAFQSAJTQANSS2mIAFNABJKLcRxKcuNEDhX3EwfkNgQJwe0ABXHHRlPLRBaDgmgODQ8Mi0ADERgA0EuKSYzsTugHo3tGg0bRgAD0IBwggXgMACP2++FAaFAMDApGqtX2-lqNTgaRAcIRaIazVaHW6fRa7S60y4mBKIDQUjgOGCwAZjKZDLJnEOUDgByYLHYMxmAD12XB+OJKH90HQ+eTijxqYJuay0ABffZCrkAZjY+wFQpFpDFu210spcqY6sVKpmBWo3JIVoYNvVdq4hDAVHlLAAtAA+G1MZ2cV3us1oH2OogqoPUGih33MIS2qO0WM2hPqoA)?

**Comment by jylertones** [comment]
Thank you @soerendomroes - I'm sorry it took me so long to see your responses. I'll try these out, they look promising.
